modification_request.go

```diff
@@ -0,0 +1,66 @@
+package braintree
+
+import (
+	"encoding/xml"
+)
+
+type ModificationRequest struct {
+	Amount                *Decimal `xml:"amount,omitempty"`
+	NumberOfBillingCycles int      `xml:"number-of-billing-cycles,omitempty"`
+	Quantity              int      `xml:"quantity,omitempty"`
+	NeverExpires          bool     `xml:"never-expires,omitempty"`
+}
+
+type AddModificationRequest struct {
+	ModificationRequest
+	InheritedFromID string `xml:"inherited-from-id,omitempty"`
+}
+
+type UpdateModificationRequest struct {
+	ModificationRequest
+	ExistingID string `xml:"existing-id,omitempty"`
+}
+
+type ModificationsRequest struct {
+	Add               []AddModificationRequest
+	Update            []UpdateModificationRequest
+	RemoveExistingIDs []string
+}
+
+func (m ModificationsRequest) MarshalXML(e *xml.Encoder, start xml.StartElement) error {
```

Comment:
The following 36 lines of code are to serialize the `ModificationsRequest` so that each of the `add`, `update` and `remove` tags within it have an attribute `type="array"` and to hide this as much as possible from the developer using the struct. The attribute is required otherwise the Braintree API returns a 403, even if the data within the tag is correct. I considered three approaches:

1. Add additional structs for `Add`, `Update`, `RemoveExistingIDs` that contains the respective arrays and an XMLAttr that serialized to `type` and that the developer had to manually set to the string `"array"`. This seemed undesirable since the developer had to set the field manually.
2. Add additional structs for `Add`, `Update`, `RemoveExistingIDs` that contains the respective arrays and a field that was a type we defined that automatically serialized itself as `type="array"` on it's enclosing struct. This seemed better since the developer did not have to set the field to anything and could not change the value that it was serialized to, but it was still undesirable because it exposes details about the serialization of the object to the developer unnecessarily.
3. Do what is done here and manually serialize the struct adding the attribute to each of the `add`, `update`, and `remove` tags. I think this is the solution we should go with, however it adds considerably more code to the repo. 36 lines is a lot for something that is largely builtin. It can be shortened a little with reflection but the code became less clear.

Any tips on how I can simplify this are much appreciated!

_Note: There is another approach I attempted where we do (2) but without the additional layer of structs. This isn't possible in Go right now because while tags can be flattened in the struct and then serialized out to a nested hierarchy, attributes cannot._